### PR TITLE
Upgraded to Pylint 1.5.x

### PR DESCRIPTION
**Pull request by clintonb**
- Updated Pylint requirement
- Removed obsolete configuration options

### CODE REVIEW

**Comment by clintonb** [comment]
Ideally, I'd like to resolve #22 before merging this.

File: setup.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 setup(
     name='edx-lint',
-    version='0.3.2',
+    version='0.4.2',
```
Comment (nedbat) [inline setup.py]: This should be 0.4.0, right?

File: setup.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
     },
 
     install_requires=[
-        'pylint==1.4.4',
+        'pylint>=1.5.0,<1.6.0',
```
Comment (nedbat) [inline setup.py]: Currently, edx-platform only gets pylint installed because of this dependency.  Developers have complained in the past of the unpredictability of the differences between running pylint locally and on CI machines.  I think it would be best to keep this precisely pinned to reduce variation.

File: setup.py (hunk 1 above)
Comment (clintonb) [inline setup.py]: Fixed.

File: setup.py (hunk 2 above)
Comment (clintonb) [inline setup.py]: Pinned to 1.5.0.

Comment (clintonb) [inline setup.py]: @nedbat this just caused problems for me (#24). If specific projects have issues with versions of pylint, those projects should pin the version. Tying this library to a specific version, however, is problematic for those who want to use other versions.

**Comment by nedbat** [comment]
:+1:

**Comment by clintonb** [comment]
I am going to merge this and create a separate PR to resolve #22. This is due to the fact that we cannot test Python 3.5 without Pylint>=1.5.1. Also, the Travis changes may require a deeper review.

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 coverage
-six
+logilab-common
```
Comment (nedbat) [inline dev-requirements.txt]: Why is this here?

File: dev-requirements.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
 coverage
-six
```
Comment (nedbat) [inline dev-requirements.txt]: Don't we still use six?

File: edx_lint/tamper_evident.py [hunk 5]
```diff
@@ -53,18 +53,16 @@ def validate(self):
...
-        start_last_line = text.rfind(u"\n", 0, -1)
+        start_last_line = text.rfind("\n", 0, -1)
         if start_last_line == -1:
             return False
 
         original_text = text[:start_last_line+1]
         last_line = text[start_last_line+1:]
 
-        expected_hash = hashlib.sha1(original_text).hexdigest().encode("ascii")
+        expected_hash = hashlib.sha1(original_text.encode("utf-8")).hexdigest()
```
Comment (nedbat) [inline edx_lint/tamper_evident.py]: This doesn't look right to me: on Py 2, original_text will be bytes already, so using .encode will raise an exception if there are non-ASCII characters.  My bad: there are no tests of non-ascii characters.

File: dev-requirements.txt (hunk 4 above)
Comment (nedbat) [inline dev-requirements.txt]: Oops, six is in setup.py now.  Sorry.

File: edx_lint/tamper_evident.py (hunk 5 above)
Comment (nedbat) [inline edx_lint/tamper_evident.py]: Wait, the code explicitly restrict the contents to ascii? weird.

**Comment by clintonb** [comment]
@nedbat Please comment on #26 regarding Travis, six, etc.
